core/src/main/scala/org/bitcoins/core/psbt/PSBTMap.scala

```diff
@@ -582,26 +571,32 @@ case class InputPSBTMap(elements: Vector[InputPSBTRecord])
       case _: WitnessScriptPubKey =>
         filterRecords(NonWitnessUTXOKeyId) :+ WitnessUTXO(transactionOutput)
       case _: P2SHScriptPubKey =>
-        if (
-          redeemScriptOpt.isDefined && redeemScriptOpt.get.redeemScript
-            .isInstanceOf[WitnessScriptPubKey]
-        ) {
-          filterRecords(NonWitnessUTXOKeyId) :+ WitnessUTXO(transactionOutput)
+        if (redeemScriptOpt.isDefined) {
+          val redeemScript = redeemScriptOpt.get.redeemScript
+          if (
+            redeemScript.isInstanceOf[WitnessScriptPubKey] && !redeemScript
+              .isInstanceOf[WitnessScriptPubKeyV0]
+          ) {
+            filterRecords(NonWitnessUTXOKeyId) :+ WitnessUTXO(transactionOutput)
+          } else {
+            elements
+          }
         } else {
           elements
         }
-      case _: P2PKHScriptPubKey | _: P2PKScriptPubKey |
-          _: P2PKWithTimeoutScriptPubKey | _: MultiSignatureScriptPubKey |
-          EmptyScriptPubKey | _: LockTimeScriptPubKey |
-          _: NonStandardScriptPubKey | _: WitnessCommitment |
-          _: ConditionalScriptPubKey =>
+      case _: WitnessScriptPubKeyV0 | _: P2PKHScriptPubKey |
```

Comment:
This new case will never be reached (`WitnessScriptPubKey` case needs to move down)